cpg_workflows/stages/realign_genotype_with_dragen.py

```diff
@@ -142,90 +140,137 @@ def queue_jobs(self, sequencing_group: SequencingGroup, inputs: StageInput) -> S
 
-        align_genotype_job: PythonJob = get_batch().new_python_job(
-            name='AlignGenotypeWithDragen',
-            attributes=(self.get_job_attrs() or {}) | {'tool': 'Dragen'},
-        )
-        align_genotype_job.image(image=image_path('ica'))
         outputs = self.expected_outputs(sequencing_group=sequencing_group)
-        pipeline_call = align_genotype_job.call(
-            run_align_genotype_with_dragen.run,
-            ica_fids_path=inputs.as_path(target=sequencing_group, stage=UploadDataToIca),
-            analysis_output_fid_path=inputs.as_path(target=sequencing_group, stage=PrepareIcaForDragenAnalysis),
-            dragen_ht_id=dragen_ht_id,
-            cram_reference_id=cram_reference_id,
-            dragen_pipeline_id=dragen_pipeline_id,
-            user_tags=user_tags,
-            technical_tags=technical_tags,
-            reference_tags=reference_tags,
-            user_reference=user_reference,
-            api_root=ICA_REST_ENDPOINT,
-        ).as_json()
-        get_batch().write_output(
-            pipeline_call,
-            str(outputs),
-        )
-        return self.make_outputs(
-            target=sequencing_group,
-            data=outputs,
-            jobs=align_genotype_job,
-        )
-
-
-@stage(analysis_type='dragen_align_genotype', required_stages=[PrepareIcaForDragenAnalysis, AlignGenotypeWithDragen])
-class MonitorAlignGenotypeWithDragen(SequencingGroupStage):
-    def expected_outputs(self, sequencing_group: SequencingGroup) -> cpg_utils.Path:
-        sg_bucket: cpg_utils.Path = sequencing_group.dataset.prefix()
-        return sg_bucket / GCP_FOLDER_FOR_RUNNING_PIPELINE / f'{sequencing_group.name}_pipeline_success.json'
 
-    def queue_jobs(self, sequencing_group: SequencingGroup, inputs: StageInput) -> StageOutput | None:
+        stage_jobs: list[BashJob | PythonJob] = []
+
+        if (
+            config_retrieve(['ica', 'pipelines', 'cancel_cohort_run'], False)
+            and cpg_utils.to_path(outputs['pipeline_id']).exists()
+        ):
+            # cancel if requested
+            logging.info('Cancelling pipeline run')
+            cancel_job = get_batch().new_python_job(
+                name='CancelIcaPipelineRun',
+                attributes=(self.get_job_attrs() or {}) | {'tool': 'Dragen'},
+            )
+            cancel_job.image(image=image_path('ica'))
+            cancel_pipeline_result = cancel_job.call(
+                cancel_ica_pipeline_run.run,
+                ica_pipeline_id_path=str(outputs['pipeline_id']),
+                api_root=ICA_REST_ENDPOINT,
+            ).as_json()
+            logging.info(f'Pipeline cancelled: {cancel_pipeline_result}')
+            get_batch().write_output(
+                cancel_pipeline_result,
+                str(
+                    sg_bucket
+                    / GCP_FOLDER_FOR_RUNNING_PIPELINE
+                    / f'{sequencing_group.name}_pipeline_cancelled_at{slugify(str(datetime.now()))}.json',
```

Comment:
This file includes the datetime, so it's guaranteed to be unique, but doesn't include the pipeline ID which was cancelled. 

If you swap this to be `f'{sequencing_group.name}_pipeline_{pipeline_id}_cancelled.json'` then you can still know the datetime from the file metadata, but you can also check for existence of this file (don't attempt to cancel the same workflow ID twice)